generation/maisi/scripts/compute_fid_2-5d_ct.py

```diff
@@ -0,0 +1,655 @@
+
+        feature_image_xy = feature_network.forward(images_2d)
+        feature_image_xy = spatial_average(feature_image_xy, keepdim=False)
+        if xy_only:
+            return feature_image_xy, None, None
+
+        # ---------------------- YZ-plane slicing along H ----------------------
+        if center_slices:
+            start_h = int((1.0 - center_slices_ratio) / 2.0 * H)
+            end_h = int((1.0 + center_slices_ratio) / 2.0 * H)
+            slices = torch.unbind(image[:, :, start_h:end_h:sample_every_k, :, :], dim=2)
+        else:
+            slices = torch.unbind(image, dim=2)
+
+        if drop_empty:
+            mapping_index = drop_empty_slice(slices, empty_threshold)
+        else:
+            mapping_index = [True for _ in range(len(slices))]
+
+        images_2d = torch.cat(slices, dim=0)
+        images_2d = radimagenet_intensity_normalisation(images_2d)
+        images_2d = images_2d[mapping_index]
+
+        feature_image_yz = feature_network.forward(images_2d)
+        feature_image_yz = spatial_average(feature_image_yz, keepdim=False)
+
+        # ---------------------- ZX-plane slicing along W ----------------------
+        if center_slices:
+            start_w = int((1.0 - center_slices_ratio) / 2.0 * W)
+            end_w = int((1.0 + center_slices_ratio) / 2.0 * W)
+            slices = torch.unbind(image[:, :, :, start_w:end_w:sample_every_k, :], dim=3)
+        else:
+            slices = torch.unbind(image, dim=3)
+
+        if drop_empty:
+            mapping_index = drop_empty_slice(slices, empty_threshold)
+        else:
+            mapping_index = [True for _ in range(len(slices))]
+
+        images_2d = torch.cat(slices, dim=0)
+        images_2d = radimagenet_intensity_normalisation(images_2d)
+        images_2d = images_2d[mapping_index]
+
+        feature_image_zx = feature_network.forward(images_2d)
+        feature_image_zx = spatial_average(feature_image_zx, keepdim=False)
+
+    return feature_image_xy, feature_image_yz, feature_image_zx
+
+
+def pad_to_max_size(tensor: torch.Tensor, max_size: int, padding_value: float = 0.0) -> torch.Tensor:
+    """
+    Zero-pad a 2D feature map or other tensor along the first dimension to match a specified size.
+
+    Args:
+        tensor (torch.Tensor): The feature tensor to pad.
+        max_size (int): Desired size along the first dimension.
+        padding_value (float): Value to fill during padding.
+
+    Returns:
+        torch.Tensor: Padded tensor matching `max_size` along dim=0.
+    """
+    pad_size = [0, 0] * (len(tensor.shape) - 1) + [0, max_size - tensor.shape[0]]
+    return F.pad(tensor, pad_size, "constant", padding_value)
+
+
+def main(
+    data0_dataroot: str = "path/to/datasetA",
+    data0_filelist: str = "path/to/filelistA.txt",
+    data0_folder: str = "datasetA",
+    data1_dataroot: str = "path/to/datasetB",
+    data1_filelist: str = "path/to/filelistB.txt",
+    data1_folder: str = "datasetB",
```

Comment:
I would suggest directly using `real` and `synth` in the argument name to make it more clear. Such as `data0_dataroot` → `real_dataset_root`, `data1_dataroot` → `synth_dataset_root`.
Also suggest rename `data0_folder` to `real_features_dir` or `real_dataset_output_name `

Reply:
Thanks for suggestion. Changes have been made.